core/arch/arm/plat-mediatek/main.c

```diff
@@ -38,6 +38,14 @@
 
 static void main_fiq(void);
 
+register_phys_mem(MEM_AREA_IO_NSEC,
+		  ROUNDDOWN(CONSOLE_UART_BASE, CORE_MMU_DEVICE_SIZE),
```

Comment:
that's right, and i don't really like that. 
Some platform may need to use fine mapping for iomem, and one must know that `MEM_AREA_IO_SEC/_NSEC` will widely map the address range.
Whatever, this shall be treated later.

Regarding current change, I did not want this change to modify the implementation related to code I do not own. However, considering the current implementation of the core, I agree with you. As Linaro is the MTK8713 maintainer and your asking, ok remove these rounding.